### DISCUSSION TITLE
reset ScrollArea scroll position (set it to the top)?

### DISCUSSION THREAD

**Discussion by MagicGrapix** [emilk/egui#4111] [Q&A] [answered]
Hello,

I'm wondering whether it is possible to reset the scroll area position to "0" scroll.
I could create the ScrollArea as `egui::ScrollArea::new([false, true]).scroll_offset(Vec2::ZERO)` and this would match the intended effect. However, the scrolling is then obviously locked. I tried calling  `egui::ScrollArea::new([false, true]).scroll_offset(Vec2::ZERO)` only when I want to reset the scroll and  `egui::ScrollArea::new([false, true])` otherwise but this does not work, the scroll remains untouched. 

Thanks a lot in advance.
Quentin

**Comment by YgorSouza** [comment, marked as answer]
The Scrolling example in the demo app shows how to do that.

https://github.com/emilk/egui/blob/e8af6f38fcebcc889963e86c4a3ccebe97db1908/crates/egui_demo_lib/src/demo/scrolling.rs#L290-L305

**Reply by MagicGrapix** [reply]
Thanks a lot, this is working.

**Comment by 13k** [comment]
Correct if I'm wrong, but my understanding is that when using the `persistence` feature, egui will persist **all** scroll positions currently in Memory and restore them on every launch.

If the above is correct, then if I want to reset scroll positions, I need to keep state if it's the "first run" and either clear the context's Memory (by "reverse-engineering" the correct persistent `Id` - which is implementation detail of `egui::containers::scroll_area::State`) or undo the scroll with `Ui::scroll_to_cursor` (which will cause "flash of original content"). And do this for **all** `ScrollArea`s. Kinda nitpicking here, but not really: that feels a lot of wasted productivity (lots of boilerplate), processing (undo things that were done automatically by default), memory (extra bookkeeping) and storage (bloating the persisted state file).

Is there a way to disable persistence of scroll positions for individual `ScrollArea`s without disabling `persistence` entirely (I'm using it to save settings and window position)?

As a more general note, IMO, it's very counter-intuitive UX that a freshly started app will, by default, show all scrollable content at previously scrolled positions. Eg.: if my settings view is scrollable and the user scrolls to the bottom, updates a setting, closes the app and then relaunches it and goes to settings again, they will see the bottom of the content. Or if a widget shows browsable content, sorted by most recent, and user scrolls to the bottom, the persisted scroll position will jump to the old stuff on relaunch. I don't think I've ever used any application that does this by default.

Scrolling seems too ephemeral a concept to have persistent state enabled by default for all instances. Shouldn't this feature be opt-in on an individual basis?
